packages/backend-plugin-manager/api-report.md

```diff
@@ -0,0 +1,199 @@
+  // (undocumented)
+  moduleLoader?(logger: LoggerService): ModuleLoader;
+}
+
+// @public (undocumented)
+export const dynamicPluginsFeatureDiscoveryServiceFactory: () => ServiceFactory<
+  FeatureDiscoveryService,
+  'root'
+>;
+
+// @public (undocumented)
+export const dynamicPluginsServiceFactory: (
+  options?: DynamicPluginsFactoryOptions | undefined,
+) => ServiceFactory<BackendPluginProvider, 'root'>;
+
+// @public (undocumented)
+export const dynamicPluginsServiceRef: ServiceRef<
+  BackendPluginProvider,
+  'root'
+>;
+
+// @public (undocumented)
+export interface FrontendDynamicPlugin extends BaseDynamicPlugin {
+  // (undocumented)
+  platform: 'web';
+}
+
+// @public (undocumented)
+export function isBackendDynamicPluginInstaller(
+  obj: any,
+): obj is BackendDynamicPluginInstaller;
+
+// @public @deprecated (undocumented)
+export interface LegacyBackendPluginInstaller {
+  // (undocumented)
+  catalog?(builder: CatalogBuilder, env: LegacyPluginEnvironment): void;
+  // (undocumented)
+  events?(
+    eventsBackend: EventsBackend,
+    env: LegacyPluginEnvironment,
+  ): HttpPostIngressOptions[];
+  // (undocumented)
+  kind: 'legacy';
+  // (undocumented)
+  permissions?: {
+    policy?: PermissionPolicy;
+  };
+  // (undocumented)
+  router?: {
+    pluginID: string;
+    createPlugin(env: LegacyPluginEnvironment): Promise<Router>;
+  };
+  // (undocumented)
+  scaffolder?(env: LegacyPluginEnvironment): TemplateAction<any>[];
+  // (undocumented)
+  search?(
+    indexBuilder: IndexBuilder,
+    schedule: TaskRunner,
+    env: LegacyPluginEnvironment,
+  ): void;
+}
+
+// @public @deprecated (undocumented)
+export type LegacyPluginEnvironment = {
+  logger: Logger;
+  cache: PluginCacheManager;
+  database: PluginDatabaseManager;
+  config: Config;
+  reader: UrlReader;
+  discovery: PluginEndpointDiscovery;
+  tokenManager: TokenManager;
+  permissions: PermissionEvaluator;
+  scheduler: PluginTaskScheduler;
+  identity: IdentityApi;
+  eventBroker: EventBroker;
+  pluginProvider: BackendPluginProvider;
+};
+
+// @public (undocumented)
+export interface ModuleLoader {
+  // (undocumented)
+  bootstrap(backstageRoot: string, dynamicPluginPaths: string[]): Promise<void>;
+  // (undocumented)
+  load(id: string): Promise<any>;
+}
+
+// @public (undocumented)
+export interface NewBackendPluginInstaller {
+  // (undocumented)
+  install(): BackendFeature | BackendFeature[];
+  // (undocumented)
+  kind: 'new';
+}
+
+// @public (undocumented)
+export class PluginManager implements BackendPluginProvider {
+  // (undocumented)
+  addBackendPlugin(plugin: BackendDynamicPlugin): void;
+  // (undocumented)
+  get availablePackages(): ScannedPluginPackage[];
+  // (undocumented)
+  backendPlugins(): BackendDynamicPlugin[];
+  // (undocumented)
+  static fromConfig(
+    config: Config,
+    logger: LoggerService,
+    preferAlpha?: boolean,
+    mooduleLoader?: ModuleLoader,
+  ): Promise<PluginManager>;
```

Comment:
For a followup, see the [API design style guidelines](https://github.com/backstage/backstage/blob/master/STYLE.md#api-design), in particular point 10. Be very eager to use this too, it makes it much easier to avoid future breaking changes.

Noting this here but it applies to more places.

Reply:
This should be fixed by PR https://github.com/backstage/backstage/pull/21764